decryptor/postgresql/utils.go

```diff
@@ -176,6 +189,142 @@ func (p *BindPacket) Zeroize() {
 	}
 }
 
+// GetParameters extracts statement parameters from Bind packet.
+func (p *BindPacket) GetParameters() ([]base.BoundValue, error) {
+	values := make([]base.BoundValue, len(p.paramValues))
+	for i := range values {
+		encoding, err := p.parameterEncoding(i)
+		if err != nil {
+			return nil, err
+		}
+		values[i] = base.NewBoundValue(p.paramValues[i], encoding)
+	}
+	return values, nil
+}
+
+func (p *BindPacket) parameterEncoding(i int) (base.BoundValueEncoding, error) {
+	// See "Bind" description in https://www.postgresql.org/docs/current/protocol-message-formats.html
+	// If there are no formats then all values use the default: text.
+	if len(p.paramFormats) == 0 {
+		return base.BindText, nil
+	}
+	// If there is only one format then it is used for all values.
+	var format uint16
+	if len(p.paramFormats) == 1 {
+		format = p.paramFormats[0]
+	} else {
+		format = p.paramFormats[i]
```

Comment:
is it okay that we don't check that `i` not bigger than `len(p.paramFormats)`?

Reply:
Mmm... Good point. I was too focused on the special cases: zero and one format.

Invalid format indices should not happen with correct packets, but I guess it makes sense to add a check and return an error in this case.